catalog/registry/catalog-registry-cswinputtransformer/src/main/java/ddf/catalog/registry/transformer/RegistryTransformer.java

```diff
@@ -68,14 +57,19 @@ public Metacard transform(InputStream inputStream, String id)
             throws IOException, CatalogTransformerException {
 
         RegistryMetacardImpl metacard;
+
         String xml = IOUtils.toString(inputStream);
         IOUtils.closeQuietly(inputStream);
 
         try {
-            metacard = (RegistryMetacardImpl) xstream.fromXML(xml);
-        } catch (XStreamException e) {
+            metacard = (RegistryMetacardImpl) unmarshal(IOUtils.toInputStream(xml));
```

Comment:
This doesn't look right to me. For one we're just doing IOUtils.toString() up above, which is extremely dangerous. Then we're going from the string back to a stream. Why not just dump the stream to something like http://google.github.io/guava/releases/19.0/api/docs/com/google/common/io/FileBackedOutputStream.html and then create as many input streams off of that as you want. Then we can protect against very large XML blowing up our heap.